app/src/main/java/com/limelight/binding/input/trackpad/Core/TrackpadFramework.java

```diff
@@ -0,0 +1,786 @@
+
+    public static final int KEY_DOWN = KeyEvent.ACTION_DOWN;
+    public static final int KEY_UP = KeyEvent.ACTION_UP;
+    public static final int BUTTON_LEFT = 0;
+    public static final int BUTTON_RIGHT = 1;
+    public static final int BUTTON_MIDDLE = 2;
+    public static final int BUTTON_X1 = 3;
+    public static final int BUTTON_X2 = 4;
+
+    public static final byte MODIFIER_SHIFT = 0x01;
+    public static final byte MODIFIER_CTRL = 0x02;
+    public static final byte MODIFIER_ALT = 0x04;
+    public static final byte MODIFIER_WIN = 0x08;
+
+    public static String getVersion(){
+        String Version;
+        int majorVersion = 1;
+        int minorVersion = 0;
+        int revisions = 0;
+        Version = majorVersion+"."+minorVersion+"."+revisions;
+        return Version;
+    }
+
+    public static String directionToString(int direction){
+        String sDirection = "";
+        switch (direction){
+            case KEY_DOWN:
+                sDirection = "KEY_DOWN";
+                break;
+            case KEY_UP:
+                sDirection = "KEY_UP";
+                break;
+        }
+
+        return sDirection;
+    }
+
+    public static String buttonToString(int button){
+        String sDirection = "";
+        switch (button){
+            case BUTTON_LEFT:
+                sDirection = "BUTTON_LEFT";
+                break;
+            case BUTTON_RIGHT:
+                sDirection = "BUTTON_RIGHT";
+                break;
+            case BUTTON_MIDDLE:
+                sDirection = "BUTTON_MIDDLE";
+                break;
+            case BUTTON_X1:
+                sDirection = "BUTTON_X1";
+                break;
+            case BUTTON_X2:
+                sDirection = "BUTTON_X2";
+                break;
+        }
+        return sDirection;
+    }
+
+
+    private static final short autoMoveWait = 500;
+    private static final short autoMoveInterval = 8;
+    private static final short autoMoveSense = 3;
+
+    private static final short moveToClickThreshold = 200;
+    private static final short clickToDragThreshold = 200;
+    private static final short dragToDoubleClickThreshold = 200;
+
+    private Timer buttonTimer;
+
+    private static short autoMoveSpaceThreshold = 100;
+
+    public TrackpadFramework(TrackpadListener listener)
+    {
+        trackpadListener = listener;
+        String deviceHead = Build.MODEL.substring(0,5);
+
+        Log.i(TAG,"DeviceName : "+ deviceHead);
+
+        if("SM-X9".equals(deviceHead)){ // Tab S8 Ultra
+            maxXThreshold = 1763;
+            minXThreshold = 0;
+            maxYThreshold = 1599;
+            minYThreshold = 528;
+            autoMoveSpaceThreshold = 100;
+        }else if("SM-T9".equals(deviceHead)||"SM-X8".equals(deviceHead)){ // Tab S7-S8 PLUS
+            maxXThreshold = 1559;
+            minXThreshold = 0;
+            maxYThreshold = 1599;
+            minYThreshold = 780;
+            autoMoveSpaceThreshold = 100;
```

Comment:
Can we query these values using `InputDevice.getMotionRange()` rather than having to hardcode them?

Reply:
"InputDevice.getMotionRange()" does not return the correct value.

For the trackpad on the Tab S series
The actual range is "528 ~ 1599", but the range reported by the device information is "0 ~ 1847".

I'll have to discuss this with Samsung, but once I had to hardcode it to use Automove.